### PR TITLE
RAF Fix and improves

### LINKED ISSUE #59: RAF Module same ID possible
It's currently possible to recruit ones own account, hence same recruiter and recruit id. I think we need to prevent this from the website.
"You can not recruit your own account."

Also, as a Feature request, the site should check for the last IP of the accounts in the auth.account database and prevent a RAF bind, if they match and are not nil (never logged in before).
"You can not recruit accounts on the same IP."

The check for same IP abuse is a config flag in the Lua. It would be a plus, If it was a config flag in the acore_cms module, too. The default can be "on".

### PR DESCRIPTION
**Pull request by nicoaravena**
This PR add a lot of graphic improves, proposing the following changes:
- adding bootstrap to wp-admin (CHECK A FEW VIEWS).
- add bootstrap components to some settings views.
- improves in RAF messages and minor UI bugs
- also add a new feature to add an optional simple IP check that validates the site user IP and account last ip with the last ip of the recruiter.

This PR close #59.

### CODE REVIEW

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php [hunk 1]
```diff
@@ -33,16 +33,18 @@ public function getHomeRender() {
...
-            <li><a href="https://www.azerothcore.org/wiki/">Wiki</a></li>
-            <li><a href="https://salt.bountysource.com/checkout/amount?team=azerothcore">Donations</a></li>
+        <div class="card w-100">
+        <p class="fs-6"><b>Welcome to AzerothCore WP Plugin.</b></p>
+        <p class="fs-6">Please take a look to the following links before continue:</p>
+        <ul class="list-unstyled">
+            <li><a href="https://www.azerothcore.org/" target="_blank">Project Homepage</a></li>
+            <li><a href="https://github.com/AzerothCore/" target="_blank">Github Repositories</a></li>
+            <li><a href="https://www.azerothcore.org/wiki/" target="_blank">Wiki</a></li>
+            <li><a href="https://www.paypal.com/donate/?hosted_button_id=L69ANPSR8BJDU" target="_blank">Sponsor</a></li>
```
Comment (Helias) [inline srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php:43]: good idea

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php [hunk 2]
```diff
@@ -33,16 +33,18 @@ public function getHomeRender() {
...
-
-        <p>Welcome to AzerothCore WP Plugin.</p>
-        <p>Please take a look to the following links before continue:</p>
-        <ul>
-            <li><a href="https://www.azerothcore.org/">Project Homepage</a></li>
-            <li><a href="https://github.com/AzerothCore/">Github Repositories</a></li>
-            <li><a href="https://www.azerothcore.org/wiki/">Wiki</a></li>
-            <li><a href="https://salt.bountysource.com/checkout/amount?team=azerothcore">Donations</a></li>
+        <div class="card w-100">
+        <p class="fs-6"><b>Welcome to AzerothCore WP Plugin.</b></p>
```
Comment (Helias) [inline srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php:37]: fix the indentation inside the div

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php [hunk 3]
```diff
@@ -33,16 +33,18 @@ public function getHomeRender() {
...
+        <p class="fs-6">Please take a look to the following links before continue:</p>
+        <ul class="list-unstyled">
+            <li><a href="https://www.azerothcore.org/" target="_blank">Project Homepage</a></li>
+            <li><a href="https://github.com/AzerothCore/" target="_blank">Github Repositories</a></li>
+            <li><a href="https://www.azerothcore.org/wiki/" target="_blank">Wiki</a></li>
+            <li><a href="https://www.paypal.com/donate/?hosted_button_id=L69ANPSR8BJDU" target="_blank">Sponsor</a></li>
+            <li><a href="https://salt.bountysource.com/checkout/amount?team=azerothcore" target="_blank">Donations</a></li>
         </ul>
-        <p>This project is fully open-source.</p>
+        <p class="fs-6">This project is fully open-source.</p>
```
Comment (Helias) [inline srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php:46]: it is ❤️

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php [hunk 4]
```diff
@@ -64,6 +66,13 @@ public function getSettingsRender() {
         ?>
 
         <form name="form-acore-settings" method="post" action="">
+
+        <div class="card p-0">
```
Comment (Helias) [inline srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php:70]: you could improve the indentation here, we should import the plugin prettier to make it automatically on-save

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php [hunk 5]
```diff
@@ -394,6 +427,18 @@ public function getElunaSettingsRender() {
                     },
                 })
             });
+            jQuery('#eluna_recruit_a_friend').on('change', function() {
+                if (this.value == '1') {
+                    jQuery('#eluna_raf_config').show();;
+                } else {
+                    jQuery('#eluna_raf_config').hide();
+                }
```
Comment (Helias) [inline srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.view.php:435]: could we use `jQuery('#eluna_raf_config').toggle();` ?

moreover tehre are two `;;` in `.show();;`

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/UserPanel/User.controller.php [hunk 6]
```diff
@@ -22,56 +22,78 @@ public function showRafProgress() {
...
+                        $activeUserIp = "";
+                        if ( ! empty( $_SERVER['HTTP_CLIENT_IP'] ) ) {
+                        //check ip from share internet
+                        $activeUserIp = $_SERVER['HTTP_CLIENT_IP'];
+                        } elseif ( ! empty( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) {
+                        //to check ip is pass from proxy
+                        $activeUserIp = $_SERVER['HTTP_X_FORWARDED_FOR'];
+                        } else {
+                        $activeUserIp = $_SERVER['REMOTE_ADDR'];
+                        }
```
Comment (Helias) [inline srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/UserPanel/User.controller.php:60]: nice, may you could improve the indentation
I am wondering what IDE you use, may you mixed tabs and spaces and you don't see the indentation like github does. Are you using atom?

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/UserPanel/User.view.php [hunk 7]
```diff
@@ -203,15 +203,16 @@ public function getRafProgressRender($rafPersonalInfo, $rafPersonalProgress, $ra
             $maxRecruitDatetime = (new \DateTime($user->get("user_registered")))->modify('+7days');
             ?>
             <div class="notice notice-info w50">
-                <p>Hey! You have not been recruited by anyone.</p>
                 <?php if ($maxRecruitDatetime >= (new \DateTime())) { ?>
                 <p>You still have until <b><?php echo $maxRecruitDatetime->format('Y-m-d H:i'); ?> [server time]</b> to be recruited by a friend, enter his username here: </p>
                 <form method="post">
                     <p>
-                        <input type="text" name="recruited" value="" placeholder="Recruiter code" size="20" >
+                        <input type="text" name="recruited" value="" placeholder="Recruiter code" size="20" required />
```
Comment (Helias) [inline srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/UserPanel/User.view.php:210]: nice catch

**Comment by Helias** [comment]
this is marvellous, I commented only small things for the rest LGTM.

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/UserPanel/User.controller.php (hunk 6 above)
Comment (nicoaravena) [reply srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/UserPanel/User.controller.php:60]: That's weird, i'm using vscode and all looked good, will take a look again

Comment (Helias) [reply srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/UserPanel/User.controller.php:61]: convert tabs into space from the settings

**Pull request merged**